provision/aws/rosa_cluster_reaper.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+set -e
+
+if [ -f ./.env ]; then
+  source ./.env
+fi
+
+echo "Starting the ROSA cluster reaper at $(date -uIs)"
+
+# Get a list of running ROSA clusters
+CLUSTERS=$(rosa list clusters --output json | jq -r '.[] | select((.state == "ready") or (.state == "pending") or (.state == "waiting") or (.state == "error")) | .name')
+
+if [ -z "$CLUSTERS" ]; then echo "Didn't find any ready state clusters"; exit 1; fi
+
+# Loop through each cluster
+for CLUSTER_NAME in $CLUSTERS; do
+  echo "Checking Cluster: $CLUSTER_NAME"
+  export CLUSTER_NAME=$CLUSTER_NAME
+  ./rosa_oc_login.sh
+
+  #Check if the 'keepalive' namespace exists in the specific cluster
+  if oc get projects | grep -q "keepalive"; then
+    echo "keepalive namespace exists in the cluster $CLUSTER_NAME., skipping deleting it."
+  else
+    echo "keepalive namespace doesn't exist in the cluster $CLUSTER_NAME., deleting it."
+
+    CLUSTER_ID=$(rosa describe cluster --cluster "$CLUSTER_NAME" | grep -oPm1 "^ID:\s*\K\w+")
```

Comment:
Can we replace lines `27-39` with `./rosa_delete_cluster.sh`?

Reply:
However, alternatively we can combine these two scripts into one and run it based on delete 1 vs delete all clusters logic, I can do that, if you want to keep these two together ? @mhajas

Reply:
> that would be problematic given in `rosa_delete_cluster.sh` the `CLUSTER_NAME` is being set this way,
> 
> ```
> CLUSTER_NAME=${CLUSTER_NAME:-$(whoami)}
> if [ -z "$CLUSTER_NAME" ]; then echo "Variable CLUSTER_NAME needs to be set."; exit 1; fi
> ```

I would say it should just work as it is already working for `./rosa_oc_login.sh` on line 19.

> However, alternatively we can combine these two scripts into one and run it based on delete 1 vs delete all clusters logic, I can do that, if you want to keep these two together ? @mhajas

Yeah, that would work too, but we would need to add additional parameters to `rosa_delete_cluster.sh`, but we could get rid of one shell script so maybe it is worth it.